src/core/ddsc/src/dds_qos.c

```diff
@@ -335,6 +335,152 @@ void dds_qset_ignorelocal (dds_qos_t * __restrict qos, dds_ignorelocal_kind_t ig
   qos->present |= QP_CYCLONE_IGNORELOCAL;
 }
 
+void dds_qset_prop (dds_qos_t * __restrict qos, const char * __restrict name, const char * __restrict value)
+{
+  bool found = false;
+  if (qos == NULL || name == NULL || value == NULL)
+    return;
+
+  if (!(qos->present & QP_PROPERTY_LIST) || qos->property.value.n == 0)
```

Comment:
There's no need to try and squeeze every last cycle out of these functions that can be squeezed out, so it makes more sense to have a general function for initialising the property list and setting ``QF_PROPERTY_LIST`` if it isn't set already. Then that function can be used for both regular & binary ones.

Then, you also don't need to treat the empty list case differently from the non-empty case, making this function (and its binary equivalent) quite a bit simpler.

Reply:
I've added an init function, and adjusted the set functions to use it.